### PR TITLE
Fix block model overlay geometry

### PR DESCRIPTION
**Pull request by zNightlord**
Small fix of this issue: [Image](https://media.discordapp.net/attachments/737872746700079235/1163524282618490900/image.png?ex=655b92ee&is=65491dee&hm=a97dea5690e99be313a6388ad21807b47484e1ca6c1629f0f8583bc4334e8d24&)

[Grass block json](https://github.com/InventivetalentDev/minecraft-assets/blob/1.20.1/assets/minecraft/models/block/grass_block.json)

Block like grass block has a small overlay on them ~~should be treated as a threat causing z fighting and clean it up~~ but let this one pass.

### CODE REVIEW

**Review by TheDuckCow — APPROVED** [review]
Thanks for the fix!

**Comment by TheDuckCow** [comment]
At face value, seems this could still cause some z-fighting issues. Seems to render fine in Eevee, but there's still a problem in cycles on render (in blender 3.6):

Screen Shot 2023-11-10 at 9 13 38 AM

My approval of the PR still stands because this is technically still better than before (we can just remove the overlay geo), but what about just the faces of "overlay" geo, or expanding it slightly to have a slight offset @zNightlord?

File: MCprep_addon/spawner/mcmodel.py [hunk 1]
```diff
@@ -367,6 +367,12 @@ def add_model(
 			)
 
 			face.normal_update()
+			
+			# Give slight offset by normal for overlay geometry
+			if face_mat == "#overlay":
+				bmesh.ops.translate(bm, verts=face.verts,
+						    vec=0.05 * face.normal)
```
Comment (TheDuckCow) [inline MCprep_addon/spawner/mcmodel.py:374]: Seems a little strong from my standpoint at 0.05, 

```suggestion
						    vec=0.02 * face.normal)
```

Not sure if that gets in the way of hte remove doubles distance, maybe drop that as well to half?

File: MCprep_addon/spawner/mcmodel.py [hunk 2]
```diff
@@ -378,7 +384,9 @@ def add_model(
 				face.material_index = materials.index(face_mat)
 
 	# Quick way to clean the model, hopefully it doesn't cause any UV issues
-	bmesh.ops.remove_doubles(bm, verts=bm.verts, dist=0.01)
+	# Ignore model has overlay geometry, causing issue
+	if not textures.get("overlay"):
+		bmesh.ops.remove_doubles(bm, verts=bm.verts, dist=0.01)
```
Comment (TheDuckCow) [inline MCprep_addon/spawner/mcmodel.py:389]: Per above suggestion, to ensure enough margin

```suggestion
		bmesh.ops.remove_doubles(bm, verts=bm.verts, dist=0.005)
```

**Review by TheDuckCow — APPROVED** [review]
I think this is good now, I'm going to add these two suggestions and merge.

**Comment by FireDasher** [comment]
this adds a depth effect that looks kinda cool but it makes holes in the mesh

**Comment by StandingPadAnimations** [comment]
@FireDasher please make a [new bug report](https://github.com/Moo-Ack-Productions/MCprep/issues/new?template=Bug-Report.yml) as opposed to commenting on a pull request merged 2 years ago (otherwise assistance becomes harder)

**Comment by FireDasher** [comment]
ok

**Pull request merged**